### PR TITLE
Register latest rules_python

### PR DESCRIPTION
**Pull request by alexeagle**
No patches are required since it natively supports bzlmod now (though not for pip deps yet)

### CODE REVIEW

File: modules/rules_python/0.10.1/MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module(
+  name = "rules_python",
+  version = "0.10.1",
+  compatibility_level = 1,
+  toolchains_to_register = ["@bazel_tools//tools/python:autodetecting_toolchain"],
+)
+
+pip_install = use_extension("@rules_python//python:extensions.bzl", "pip_install")
```
Comment (fmeum) [inline modules/rules_python/0.10.1/MODULE.bazel:8]: Nit: 
```suggestion
pip_install = use_extension("//python:extensions.bzl", "pip_install")
```

**Comment by meteorcloudy** [comment]
Looks like building "@rules_python//python/pip_install/extract_wheels/lib" is not working.

**Comment by alexeagle** [comment]
ah, simple answer, that lib subpackage was removed in https://github.com/bazelbuild/rules_python/pull/743

**Comment by alexeagle** [comment]
@fmeum PTAL, green now

**Review by fmeum — APPROVED** [review]

**Pull request merged**